### DISCUSSION TITLE
Gracefully stopping cineast during an exception in the extraction process

### DISCUSSION THREAD

**Discussion by RibinMathewChalumattu** [vitrivr/cineast#173] [Q&A] [answered]
Dear devs,

What are the best practices to stop cineast gracefully, when an exception occurs during the extraction process? Specifically, I want to revert any changes made to all tables during the extraction process.

Best,

Ribin

**Comment by lucaro** [comment, marked as answer]
Hey, vitrivr does not currently support a transactional extraction process. If you want a all-or-nothing semantic in your database, you would need to extract to a separate location, such as a json file, and import the results after the extraction completes. Is there any particular reason you are having exceptions during the extraction? This is not something that usually happens.

**Comment by silvanheller** [comment]
There's also some discussion in #114 on this topic as additional context
